Source/Core/Common/bbaserver.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QObject>
+#include <QSet>
+
+class QHostAddress;
+class QNetworkProxy;
+class QTcpServer;
+
+class BbaClient;
+class BbaDebug;
+
+class BbaServer : public QObject
+{
+  Q_OBJECT
+
+public:
+  explicit BbaServer(QObject *parent = Q_NULLPTR);
+
+  bool listen(const QHostAddress &address, quint16 port = 0);
+  bool listen(const QString &address, quint16 port = 0);
+  void close();
+
+  bool isListening() const;
+
+  void setMaxPendingConnections(int numConnections);
+  int maxPendingConnections() const;
+
+  quint16 serverPort() const;
+  QHostAddress serverAddress() const;
+
+  //QAbstractSocket::SocketError serverError() const; // can't be forward declared
+  QString errorString() const;
+
+  void pauseAccepting();
+  void resumeAccepting();
+
+#ifndef QT_NO_NETWORKPROXY
+  void setProxy(const QNetworkProxy &networkProxy);
+  QNetworkProxy proxy() const;
+#endif
+
+  BbaDebug logInfo();
+
+Q_SIGNALS:
+  void information(const QDateTime &timestamp, const QString &message);
+
+protected:
+  void timerEvent(QTimerEvent *ev);
+
+private Q_SLOTS:
+  void newConnection();
+
+private:
+  QSet<BbaClient*> m_clients;
```

Comment:
Unless this relies on something in QSet's interface that isn't in the standard library, this can be a `std::unordered_set`